### ISSUE TITLE
Dire Wolf Alpha + Meanstreet Marshal

### ISSUE THREAD

**Issue by DominusMaximus** [demilich1/metastone#302]
- If I have a Dire Wolf Alpha inbetween two Meanstreet Marshals, and my opponent plays the aoe version of Starfall, I should draw 2 cards.

Here's a nice post from Patashu explaining this mechanic https://us.battle.net/forums/en/hearthstone/topic/20419273711#post-8

> The specific mechanic is that between Phases, Aura Update (Health/Attack) runs before the Death Creation Step, and Aura Update (Other) runs after. Meaning that a minion that dies and was emitting a health aura does not update (to stop emitting its health aura) until after the following Death Phase, where on-death triggers (including Deathrattles) are resolved for all entities that just died.

Also:

> This scenario is different because Health/Attack Auras are not tied to the minion being in play or not NOR are they a trigger on death - they update every Aura Update (Health/Attack), which is before every corresponding Death Creation Step. So when an Entity with a Deathrattle and a Health Aura minion die simultaneously, the Health Aura is still in play for the subsequent Death Phase (where the Deathrattle resolves), THEN it updates and stops being emitted.

**Comment by webadict** [comment]
And I almost thought it was nearly complete. And another bug shows its face.

But, yeah, makes sense. I'll see if Auras can exist until they're killed. It _shouldn't_ be an issue, but it seems easier than it probably is.

**Comment by webadict** [comment]
Well, problem solved. Apparently, when a minion is killed, all spell triggers are killed with it, before resolving deathrattles. This makes sense, since spell triggers include things like triggered effects, so something might not fully remove itself. However, Auras aren't SPECIFICALLY like that, which makes this kind of an issue... So, I just made a special exception for Auras. Auras are now removed specifically at the end of the death phase, which is where they are also updated.
